x/interchainstaking/keeper/msg_server.go

```diff
@@ -28,6 +35,313 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+			if err != nil {
+				return &types.MsgUpdateZoneResponse{}, err
+			}
+			zone.LiquidityModule = boolValue
+
+		case types.UpdateZoneKeyUnbondingEnabled:
+			boolValue, err := strconv.ParseBool(change.Value)
+			if err != nil {
+				return &types.MsgUpdateZoneResponse{}, err
+			}
+			zone.UnbondingEnabled = boolValue
+
+		case types.UpdateZoneKeyDepositsEnabled:
+			boolValue, err := strconv.ParseBool(change.Value)
+			if err != nil {
+				return &types.MsgUpdateZoneResponse{}, err
+			}
+			zone.DepositsEnabled = boolValue
+
+		case types.UpdateZoneKeyReturnToSender:
+			boolValue, err := strconv.ParseBool(change.Value)
+			if err != nil {
+				return &types.MsgUpdateZoneResponse{}, err
+			}
+			zone.ReturnToSender = boolValue
+
+		case types.UpdateZoneKeyMessagesPerTx:
+			intVal, err := strconv.Atoi(change.Value)
+			if err != nil {
+				return &types.MsgUpdateZoneResponse{}, err
+			}
+			if intVal < 1 {
+				return &types.MsgUpdateZoneResponse{}, fmt.Errorf("invalid value for messages_per_tx: %d", intVal)
+			}
+			zone.MessagesPerTx = int64(intVal)
+
+		case types.UpdateZoneKeyAccountPrefix:
+			zone.AccountPrefix = change.Value
+
+		case types.UpdateZoneKeyIs118:
+			boolValue, err := strconv.ParseBool(change.Value)
+			if err != nil {
+				return &types.MsgUpdateZoneResponse{}, err
+			}
+			zone.Is_118 = boolValue
+
+		case types.UpdateZoneKeyConnectionID:
+			if !strings.HasPrefix(change.Value, types.ConnectionPrefix) {
+				return &types.MsgUpdateZoneResponse{}, errors.New("unexpected connection format")
+			}
+			if zone.DepositAddress != nil || zone.DelegationAddress != nil || zone.PerformanceAddress != nil || zone.WithdrawalAddress != nil {
+				return &types.MsgUpdateZoneResponse{}, errors.New("zone already intialised, cannot update connection_id")
+			}
+			if k.BankKeeper.GetSupply(ctx, zone.LocalDenom).Amount.IsPositive() {
+				return &types.MsgUpdateZoneResponse{}, errors.New("zone has assets minted, cannot update connection_id without potentially losing assets")
+			}
+
+			connection, found := k.IBCKeeper.ConnectionKeeper.GetConnection(ctx, change.Value)
+			if !found {
+				return &types.MsgUpdateZoneResponse{}, errors.New("unable to fetch connection")
+			}
+
+			clientState, found := k.IBCKeeper.ClientKeeper.GetClientState(ctx, connection.ClientId)
+			if !found {
+				return &types.MsgUpdateZoneResponse{}, errors.New("unable to fetch client state")
+			}
+
+			tmClientState, ok := clientState.(*tmclienttypes.ClientState)
+			if !ok {
+				return &types.MsgUpdateZoneResponse{}, errors.New("error unmarshaling client state")
+			}
+
+			if tmClientState.Status(ctx, k.IBCKeeper.ClientKeeper.ClientStore(ctx, connection.ClientId), k.IBCKeeper.Codec()) != ibcexported.Active {
+				return &types.MsgUpdateZoneResponse{}, errors.New("new connection client state is not active")
+			}
+
+			zone.ConnectionId = change.Value
+
+			k.SetZone(ctx, &zone)
+
+			// generate deposit account
+			portOwner := zone.ID() + ".deposit"
```

Comment:
And consts here too please

Reply:
Will use https://github.com/ingenuity-build/quicksilver/pull/502/commits/f22910740f719ccde6c6a1c6d354a96bf6cab222 once #502 is merged